### PR TITLE
Add ability to return results as json.

### LINKED ISSUE #24: Have an argument which returns JSON in `what`
This is for like an open API that lets people use it!

### PR DESCRIPTION
**Pull request by RackReaver**
- Added exporting details to README
- Updated README formatting (auto formatted)
- Added click option to `pywhat/what.py:main`
- Updated `pywhat/what.py:main` docstring
- Updated `pywhat/printer.py:Printing.print_json()` with new code
- Refactored a bit of `pywhat/printer.py:Printing`
- Added test for valid json
- Added test for empty json return

fixes #24

### CODE REVIEW

File: pywhat/printer.py [hunk 1]
```diff
@@ -98,4 +99,13 @@ def pretty_print(self, text: dict, text_input):
         """
 
     def print_json(self, text: dict):
-        return json.dumps(text, indent=4)
+        if text['Regexes'] == None:
```
Comment (piatrashkakanstantinass) [inline pywhat/printer.py:102]: Please format it all using black.🖤

File: pywhat/what.py [hunk 2]
```diff
@@ -145,9 +152,14 @@ def main(text_input, rarity, include_tags, exclude_tags, only_text, key, reverse
             print("Invalid key")
             sys.exit(1)
     identified_output = what_obj.what_is_this(text_input, only_text, key, reverse)
-
+    
     p = printer.Printing()
-    p.pretty_print(identified_output, text_input)
+    
+    if json == True:
```
Comment (piatrashkakanstantinass) [inline pywhat/what.py:158]: Just make it `if json:`

**Review by piatrashkakanstantinass — CHANGES_REQUESTED** [review]

File: pywhat/printer.py [hunk 3]
```diff
@@ -98,4 +99,13 @@ def pretty_print(self, text: dict, text_input):
...
+            self.console.print('{}')
+
+        else:
+            delete_values = ['Regex', 'plural_name', 'Tags', 'Rarity']
+            values = text['Regexes']['text'][0]['Regex Pattern']
```
Comment (piatrashkakanstantinass) [inline pywhat/printer.py:107]: It does not work for files, can you fix this somehow?🙏

**Review by piatrashkakanstantinass — CHANGES_REQUESTED** [review]

File: pywhat/printer.py (hunk 3 above)
Comment (RackReaver) [reply pywhat/printer.py:107]: Are there any other input types I should be testing? I only tested str arguments.

File: pywhat/printer.py (hunk 1 above)
Comment (RackReaver) [reply pywhat/printer.py:102]: I will get it done after making the code changes.

File: pywhat/printer.py (hunk 3 above)
Comment (bee-san) [reply pywhat/printer.py:107]: Nope! Just files and strings for this :)

Comment (piatrashkakanstantinass) [reply pywhat/printer.py:107]: +folders

**Review by bee-san — APPROVED** [review]

**Comment by bee-san** [comment]
thank you!!

**Pull request merged**